lib/Sema/TypeChecker.cpp

```diff
@@ -558,6 +558,12 @@ void swift::performTypeChecking(SourceFile &SF, TopLevelContext &TLC,
   if (SF.ASTStage == SourceFile::TypeChecked)
     return;
 
+  // Resolve condition clauses so name binding can accurately track imports.
+  {
+    SharedTimer timer("Condition resolution");
+    performConditionResolution(SF);
```

Comment:
I wasn't sure if this should be folded in with NameBinding, but I err'd on the side of caution because of the concerns about slowdown for this feature.